src/java.base/share/classes/java/lang/doc-files/ValueBased.html

```diff
@@ -31,36 +31,42 @@
 <body>
 <h1 id="ValueBased">{@index "Value-based Classes"}</h1>
 
-Some classes, such as <code>java.util.Optional</code> and
-<code>java.time.LocalDateTime</code>, are <em>value-based</em>.  Instances of a
-value-based class:
+Some classes, such as <code>java.lang.Integer</code> and
+<code>java.time.LocalDateTime</code>, are <em>value-based</em>.
+Value-based classes:
 <ul>
-    <li>are final and immutable (though may contain references to mutable
-        objects);</li>
-    <li>have implementations of <code>equals</code>,
+    <li>declare only final instance fields (though these may contain references
+        to mutable objects);</li>
+    <li>declare implementations of <code>equals</code>,
         <code>hashCode</code>, and <code>toString</code> which are computed
         solely from the instance's state and not from its identity or the state
         of any other object or variable;</li>
-    <li>make no use of identity-sensitive operations such as reference
-        equality (<code>==</code>) between instances, identity hash code of
-        instances, or synchronization on an instances's intrinsic lock;</li>
-    <li>are considered equal solely based on <code>equals()</code>, not
-        based on reference equality (<code>==</code>);</li>
-    <li>do not have accessible constructors, but are instead instantiated
-        through factory methods which make no commitment as to the identity
-        of returned instances;</li>
-    <li>are <em>freely substitutable</em> when equal, meaning that interchanging
-        any two instances <code>x</code> and <code>y</code> that are equal
-        according to <code>equals()</code> in any computation or method
-        invocation should produce no visible change in behavior.
+    <li>have instances that are <em>freely substitutable</em> when equal, meaning
+        that interchanging any two instances <code>x</code> and <code>y</code> that
+        are equal according to <code>equals()</code> produces no visible change in
+        the behavior of the class's methods;
     </li>
+    <li>perform no synchronization on an instance's intrinsic lock;</li>
+    <li>do not have (or have deprecated any) accessible constructors;</li>
+    <li>may support instance creation through factory methods that do <em>not</em>
+        promise a unique identity for each invocation—in particular, each factory
+        method must allow for the possibility that if two independently-produced
+        instances are equal according to <code>equals()</code>, they may also be
+        equal according to <code>==</code>;
```

Comment:
Is using "may" a way to allow contradiction of the previous requirements?
Otherwise, in a spec, any statement using 'may' can be ignored or removed, it does not specify any behavior either required or disallowed.

Reply:
What I'm going for is "may declare factory methods, and if they do then...". But you're right, it doesn't work as written. I'll rephrase. (Why not require a factory method? Because a class could also just define a handful of named instances, enum-style, and not expose any factory-like behavior. Boolean is almost an example of this, although it does have a 'valueOf' method, too.)